src/classes/CW_CampaignTriggerHandler.cls

```diff
@@ -0,0 +1,308 @@
+ 					newObjectMap.get(id).Stage__c = ConstantUtil.CAMP_STAGE_DRAFT;
+ 			}
+ 		}
+ 	}
+            	
+	public void OnAfterUpdate(
+		CW_Campaign__c[] oldObjects, 
+		CW_Campaign__c[] updatedObjects, 
+		Map<Id,CW_Campaign__c> oldObjectMap, 
+		Map<Id,CW_Campaign__c> updatedObjectMap
+	) {
+		// we are interested in Campaigns where the points have changed
+		// OR where the campaign was previous a child of a Roadmap (parent) and will impact points on the parent
+		List<CW_Campaign__c> campaignsWithUpdatedPoints = new List<CW_Campaign__c>();
+		Map<Id,List<CW_Campaign__c>> reparentedObjectMap = new Map<Id,List<CW_Campaign__c>>();
+		
+		// we are interested in campaigns where the stage has changed
+		Map<Id,CW_Campaign__c> campaignsWithUpdatedStage = new Map<Id,CW_Campaign__c>();
+		
+		for (CW_Campaign__c c :updatedObjects) {
+			
+			CW_Campaign__c oldObject = oldObjectMap.get(c.Id);
+			
+			// campaign has a Roadmap (parent)
+			if ( c.Roadmap__c!=null ) {
+				
+				// we are interested in campaigns where points have changed
+				// OR the campaign previously did not have a Roadmap and will now have a impact on points
+				if ( 
+					c.Points__c!=oldObject.Points__c 
+					|| ( oldObject.Roadmap__c==null && ( oldObject.Points__c!=null && oldObject.Points__c!=0 ) )
+				) {
+					campaignsWithUpdatedPoints.add(c);
+				}
+			}
+			// campaign does not have a Roadmap (parent)
+			else {
+				
+				// we are only interested in campaigns which previously had a roadmap AND will have an impact on points
+				if ( oldObject.Roadmap__c!=null && ( oldObject.Points__c!=null && oldObject.Points__c!=0 ) ) {
+					List<CW_Campaign__c> reparentedCamps = reparentedObjectMap.get(oldObject.Roadmap__c);
+					if (reparentedCamps==null) reparentedCamps = new List<CW_Campaign__c>();
+					reparentedCamps.add(c);
+					reparentedObjectMap.put(oldObject.Roadmap__c, reparentedCamps);
+				}
+			}
+			
+			// campaign stage has changed
+			if (c.Stage__c!=oldObject.Stage__c) {
+				campaignsWithUpdatedStage.put(c.Id,c);
+			}
+		}
+		
+		// if campaign previously had a roadmap, remove its campaigns points from the Roadmap (parent)
+    	if (!reparentedObjectMap.isEmpty()) updateRoadmapWhenCampaignsHaveBeenReparented(reparentedObjectMap);
+    	
+        if (!campaignsWithUpdatedPoints.isEmpty()) rollupPointsOnCampaignToRoadmap(campaignsWithUpdatedPoints);
+        
+        // if campaign stage has changed, update its child schedules with the new stage
+        if (!campaignsWithUpdatedStage.isEmpty()) updateChildSchedulesWithCampaignStage(campaignsWithUpdatedStage);
+        
+        // handle schedules
+        cascadeDeleteSchedulesOnUpdate(updatedObjects,oldObjectMap);
+    }
+    
+    public void OnAfterDelete( CW_Campaign__c[] deletedObjects, Map<Id,CW_Campaign__c> objectMap ) {
+        rollupPointsOnCampaignToRoadmap(deletedObjects);
+    }
+
+    public void OnUndelete( CW_Campaign__c[] restoredObjects, Map<Id,CW_Campaign__c> restoredObjectMap ) {
```

Comment:
Style issue: methods should use the "lowerCamelCase" notation. Suggestion: rename accordingly.